retroarch.c

```diff
@@ -11449,22 +11474,9 @@ static bool command_event_main_state(
    bool ret                    = false;
    bool push_msg               = true;
 
-   state_path[0] = msg[0]      = '\0';
-
-   if (global)
-   {
-      int state_slot             = settings->ints.state_slot;
-      const char *name_savestate = global->name.savestate;
+   msg[0]                      = '\0';
```

Comment:
@bulzipke Many thanks for updating the PR! There's just one tiny issue remaining. Because you changed:

```
state_path[0] = msg[0]      = '\0';
```
to
```
msg[0]                      = '\0';
```
`state_path` will end up unterminated in the event that `retroarch_get_current_savestate_path()` fails. In this case, anything that uses `state_path` will subsequently overflow the buffer.

So basically, in both the places that you call `retroarch_get_current_savestate_path()`, you just need to make sure that `state_path[0] = '\0'` is set first.

With this last fix, we can go ahead and merge :)

Reply:
Oh, thank you for pointing out my mistake.
I'm still unfamiliar with github, so please let me know if there's any additional work to do (e.g., rebase, etc.)